applications/RANSApplication/custom_processes/rans_apply_exact_nodal_periodic_condition_process.cpp

```diff
@@ -0,0 +1,321 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Suneth Warnakulasuriya
+//
+
+// System includes
+#include <limits>
+#include <string>
+
+// External includes
+
+// Project includes
+#include "fluid_dynamics_application_variables.h"
+#include "includes/define.h"
+#include "processes/reorder_and_optimize_modelpart_process.h"
+
+// Application includes
+
+// Include base h
+#include "rans_apply_exact_nodal_periodic_condition_process.h"
+
+namespace Kratos
+{
+RansApplyExactNodalPeriodicConditionProcess::RansApplyExactNodalPeriodicConditionProcess(
+    Model& rModel,
+    Parameters rParameters)
+: mrModel(rModel), mrParameters(rParameters)
+{
+    KRATOS_TRY
+
+    Parameters default_parameters = Parameters(R"(
+        {
+            "base_model_part_name"           : "PLEASE_SPECIFY_MODEL_PART_NAME",
```

Comment:
I think this is no longer required as it needs to be provided in the `master_model_part_name` and `slave_model_part_name`. Am I wrong?

Reply:
I am creating Periodic Conditions on the fly, and those periodic conditions will be created on the base_model_part.

Reply:
My point is to retrieve the root model part from the model part using the `GetRootModelPart` method.

Reply:
removed as discussed :)